cmd/controller-manager/app/helper_test.go

```diff
@@ -75,3 +79,63 @@ func TestIsControllerEnabled(t *testing.T) {
 	}
 
 }
+
+func TestParseClusterCIDRs(t *testing.T) {
+	type cs struct {
+		name       string
+		cidrs      string
+		ipv6Enable bool
+		valid      bool
+	}
+	tcs := []cs{
+		{
+			name:  "invalid cidr",
+			cidrs: "127.0.0.1",
+			valid: false,
+		},
+		{
+			name:  "single IPv4 cidr",
+			cidrs: "192.168.0.0/24",
+			valid: true,
+		},
+		{
+			name:  "dual cidrs without feature gate",
+			cidrs: "172.10.0.1/24,2000::/3",
+			valid: false,
+		},
+		// set the feature gate
+		{
+			name:       "dual IPv4 cidrs",
+			cidrs:      "192.168.0.0/24,172.10.0.0/24",
+			ipv6Enable: true,
+			valid:      false,
+		},
+		{
+			name:       "dual cidrs",
+			cidrs:      "172.10.0.1/24,2000::/3",
+			ipv6Enable: true,
+			valid:      true,
+		},
+		{
+			name:       "trinal cidrs",
+			cidrs:      "192.168.0.0/24,172.10.0.1/24,2000::/3",
+			ipv6Enable: true,
+			valid:      false,
+		},
+	}
+
+	wait := sync.WaitGroup{}
+	wait.Add(len(tcs))
+	for _, tc := range tcs {
```

Comment:
You can use t.Run to make this a bit cleaner
```
for _, tc := range tcs {
    t.Run(tc.name, func(t *testing.T){ 
         ....
   })
}
```
And that way you don't have to print the test name on failure since t.Run will do that for you

Reply:
(no wait group as well)